### ISSUE TITLE
Missing install of 

### ISSUE THREAD

**Issue by casati-dolibarr** [tsolucio/corebos#64]
Pls. add this code to `CreateTables.inc.php` in `function populateLinks()` before installation to enable display of 'Contact Hierarchy' in 'DetailView.tpl' (Pull request w/code follows):

```...
$contactInstance = Vtiger_Module::getInstance('Contacts');
        $contactInstance->addLink(
                'DETAILVIEWBASIC', 'LBL_ADD_NOTE', 
                'index.php?module=Documents&action=EditView&return_module=$MODULE$&return_action=DetailView&return_id=$RECORD$&parent_id=$RECORD$',
                'themes/images/bookMark.gif'
        );
        $contactInstance->addLink('DETAILVIEWBASIC', 'LBL_SHOW_CONTACT_HIERARCHY', 'index.php?module=Contactss&action=ContacttHierarchy&contactid=$RECORD$');
...```

Best regards

**Comment by joebordes** [comment]
I'll wait to see the PR but now we use coreBOS updater for this type of change, that way we avoid having to do upgrades and everybody gets the change.

http://corebos.org/documentation/doku.php?id=en:devel:corebosupdater

**Comment by casati-dolibarr** [comment]
PR ready (git push'ed). What to do next?

Changes+Additions Peter A. Gebhardt (casati-dolibarr) 2015-Dec-22 	30f0b33

Best regards

**Comment by joebordes** [comment]
The steps to make a PR are like this:
- create a branch for your change: ```git checkout -b newfeature master```
- work in the branch, implement your changes and test them
- commit to the branch
- push the changes to github
- open the branch in github web site and you will see the Pull Request button

**Comment by joebordes** [comment]
I need you to do that because the commit you have made has over 350 files, many completely changed so I don't know what you have done.
Look at most of the commits we do, they are usually small, concise, more or less one commit is one concept, fix or feature.

**Comment by casati-dolibarr** [comment]
Apparently this doesnt work:

----------------------------
root@pag-a52f:/var/www/corebos-master# git status
# On branch master
# Your branch is up-to-date with 'origin/master'.
#
# Untracked files:
#   (use "git add <file>..." to include in what will be committed)
#
#	ContactHierarchy.tpl
#	Contacts.php
#	DetailView.tpl
#	DetailViewBlockComment.tpl
#	DetailViewBlockCommentItem.tpl
#	DetailViewUI.tpl
#	ListView.php
#	ListViewController.php
#	RelatedListContents.tpl
#	UnifiedSearchAjax.tpl
#	editexpressionscript.js
#	en_us.lang.php
#	utils.php
#
nothing added to commit but untracked files present (use "git add" to track)
root@pag-a52f:/var/www/corebos-master# git add .
root@pag-a52f:/var/www/corebos-master# git status
# On branch master
# Your branch is up-to-date with 'origin/master'.
#
# Changes to be committed:
#   (use "git reset HEAD <file>..." to unstage)
#
#	new file:   ContactHierarchy.tpl
#	new file:   Contacts.php
#	new file:   DetailView.tpl
#	new file:   DetailViewBlockComment.tpl
#	new file:   DetailViewBlockCommentItem.tpl
#	new file:   DetailViewUI.tpl
#	new file:   ListView.php
#	new file:   ListViewController.php
#	new file:   RelatedListContents.tpl
#	new file:   UnifiedSearchAjax.tpl
#	new file:   editexpressionscript.js
#	new file:   en_us.lang.php
#	new file:   utils.php
#
root@pag-a52f:/var/www/corebos-master# git commit -a -m
"Changes+Additions Peter A. Gebhardt (casati-dolibarr) 2015-Dec-23"
[master a87b2e8] Changes+Additions Peter A. Gebhardt (casati-dolibarr)
2015-Dec-23
 14 files changed, 9796 insertions(+)
 create mode 100755 ContactHierarchy.tpl
 create mode 100644 Contacts.php
 create mode 100644 DetailView.tpl
 create mode 100644 DetailViewBlockComment.tpl
 create mode 100644 DetailViewBlockCommentItem.tpl
 create mode 100644 DetailViewUI.tpl
 create mode 100644 ListView.php
 create mode 100644 ListViewController.php
 create mode 100644 RelatedListContents.tpl
 create mode 100644 UnifiedSearchAjax.tpl
 create mode 100644 editexpressionscript.js
 create mode 100644 en_us.lang.php
 create mode 100644 utils.php
root@pag-a52f:/var/www/corebos-master# git push
Username for 'https://github.com': casati-dolibarr
Password for 'https://casati-dolibarr@github.com':
remote: Permission to tsolucio/corebos.git denied to casati-dolibarr.
**** fatal: unable to access 'https://github.com/tsolucio/corebos.git/':
The requested URL returned error: 403
root@pag-a52f:/var/www/corebos-master# git config -l
user.name=casati-dolibarr
user.email=pag@pag-consulting.com
push.default=simple
core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
remote.origin.url=https://github.com/tsolucio/corebos.git
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
branch.master.remote=origin
branch.master.merge=refs/heads/master
root@pag-a52f:/var/www/corebos-master#

----------

Best regards

**Comment by joebordes** [comment]
git status > On branch master
git checkout -b contacthierarchy master
git add ContactHierarchy.tpl Contacts.php {all other filenames}
git commit -m "feat(Contacts) Contact Hierarchy"
git push (you may need some additional parameters here -u)
git checkout master
now go to github and start the pull request from the branch

be patient, all this git stuff is a little complex, there is a learning curve, but you can catch on quickly and it is worth it

**Comment by joebordes** [comment]
BTW, the files shouldn't be in the root.

**Comment by casati-dolibarr** [comment]
Still no success ...

	root@pag-a52f:/var/www/corebos-master# git push
	fatal: The current branch contacthierarchy has no upstream branch.
	To push the current branch and set the remote as upstream, use

		git push --set-upstream origin contacthierarchy

	root@pag-a52f:/var/www/corebos-master# git push -u 
	fatal: The current branch contacthierarchy has no upstream branch.
	To push the current branch and set the remote as upstream, use

		git push --set-upstream origin contacthierarchy

	root@pag-a52f:/var/www/corebos-master#  git push --set-upstream origin contacthierarchy
	Username for 'https://github.com': casati-dolibarr
	Password for 'https://casati-dolibarr@github.com': 
	remote: Permission to tsolucio/corebos.git denied to casati-dolibarr.
	fatal: unable to access 'https://github.com/tsolucio/corebos.git/': The requested URL returned error: 403
	root@pag-a52f:/var/www/corebos-master# git config --global push.default current
	root@pag-a52f:/var/www/corebos-master# git push -u
	Username for 'https://github.com': casati-dolibarr
	Password for 'https://casati-dolibarr@github.com': 
	remote: Permission to tsolucio/corebos.git denied to casati-dolibarr.
	fatal: unable to access 'https://github.com/tsolucio/corebos.git/': The requested URL returned error: 403
	root@pag-a52f:/var/www/corebos-master# git config -l
	user.name=casati-dolibarr
	user.email=pag@pag-consulting.com
	push.default=current
	core.repositoryformatversion=0
	core.filemode=true
	core.bare=false
	core.logallrefupdates=true
	remote.origin.url=https://github.com/tsolucio/corebos.git
	remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
	branch.master.remote=origin
	branch.master.merge=refs/heads/master
	root@pag-a52f:/var/www/corebos-master#

**Comment by joebordes** [comment]
You can't push to coreBOS project directly you have to push to your fork.
You forked the project to your account, there you can work and modify the project as you please, once you have a branch you make a PR from your fork to the coreBOS project
In other words, do all the steps I have been indicating but on your fork.

Keep asking :-)

**Comment by casati-dolibarr** [comment]
Here we go ...

	root@pag-a52f:/var/www/corebos-master# git config -l
	user.name=casati-dolibarr
	user.email=pag@pag-consulting.com
	push.default=current
	core.repositoryformatversion=0
	core.filemode=true
	core.bare=false
	core.logallrefupdates=true
	remote.origin.url=https://github.com/casati-dolibarr/corebos.git
	remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
	branch.master.remote=origin
	branch.master.merge=refs/heads/master
	root@pag-a52f:/var/www/corebos-master# git push
	Username for 'https://github.com': casati-dolibarr
	Password for 'https://casati-dolibarr@github.com': 
	Counting objects: 27, done.
	Delta compression using up to 2 threads.
	Compressing objects: 100% (26/26), done.
	Writing objects: 100% (27/27), 14.85 KiB | 0 bytes/s, done.
	Total 27 (delta 22), reused 0 (delta 0)
	To https://github.com/casati-dolibarr/corebos.git
	 * [new branch]      contacthierarchy -> contacthierarchy

**Comment by joebordes** [comment]
Ok, this is starting to look better.
Now go to your project
Open the branches drop down
you will see your new branch
select it
you will see the "Create Pull Request" button
click on it
you will see the exact changes your are proposing me to incorporate to coreBOS
Now we can start discussing what needs to change for me to accept them

**Comment by joebordes** [comment]
I had a quick look, there are a lot of changes that I understand are not needed for the Contact Hierarchy. In fact all the changes to template files are not needed and I don't understand what those changes are about.
In the language file, the whole file is changed when I understand that you are probably just adding one label, no?
Please clean that up, so that only the relevant changes are in the PR

**Comment by casati-dolibarr** [comment]
That's how the database looks like.

Bst regards

**Comment by casati-dolibarr** [comment]
I've just decided NOT to create a pull request for Contact Hierarchy, ModComment and the custom Tab-Layout.

Neither you not me would be able to sort out the sheer amount of diffs created from moving the required code from vt540 to cb-dev rolling release.

Using a custom (fixed) combination of Show/Hide of Blocks, I'm able to reproduce nearly all positive workflow enhancements.

**Comment by casati-dolibarr** [comment]
I've just decided NOT to create a pull request for ModComment and the custom Tab-Layout.

Neither you not me would be able to sort out the sheer amount of diffs created from moving the required code from vt540 to cb-dev rolling release.

Using a custom (fixed) combination of Show/Hide of Blocks, I'm able to reproduce nearly all positive workflow enhancements.
